src/licensedcode/plugin_license.py

```diff
@@ -116,3 +119,57 @@ def get_scanner(
             license_text_diagnostics=license_text_diagnostics,
             license_url_template=license_url_template
         )
+
+    def process_codebase(self, codebase, **kwargs):
+        
+        if codebase.has_single_resource:
+            return
+
+        for resource in codebase.walk(topdown=False):
+            match_reference_license(resource,codebase)
+
+
+def match_reference_license(resource, codebase):
```

Comment:
This would need unit tests

Reply:
Ig I have already added some test cases!

Reply:
The tests currently added are whole scans and not unittests for just `match_reference_license` function

Reply:
You can create [mock codebase](https://github.com/nexB/scancode-analyzer/blob/main/tests/test_analyzer_plugin.py#L200) from JSONs and also create mock resources directly like [this](https://github.com/nexB/scancode-analyzer/blob/main/tests/test_analyzer_plugin.py#L175). This is the only way to get resource/codebases outside of scans. Now use these to test various  cases of the function `match_reference_license`, where they are able to follow and update.